frontend/viewer/src/lib/troubleshoot/TroubleshootDialog.svelte

```diff
@@ -5,10 +5,30 @@
   import { useFwLiteConfig, useTroubleshootingService } from '$lib/services/service-provider';
   import { AppNotification } from '$lib/notifications/notifications';
   import { t } from 'svelte-i18n-lingui';
+  import {QueryParamStateBool} from '$lib/utils/url.svelte';
+  import {watch} from 'runed';
+
+  let {
+    open = $bindable(),
+  } = $props<{
+    open: boolean;
+  }>();
+
+  const openQueryParam = new QueryParamStateBool('troubleshootDialogOpen', true);
+
+  watch(() => open, (newOpen) => {
+    openQueryParam.current = newOpen;
+  });
+  // 	Detect when the dialog is closed via back-navigation
+  watch(() => openQueryParam.current, () => {
```

Comment:
after the issues we just had with similar code in lexbox I don't love this dual watch sync, but I'm not clear if there's a better option outside of exposing a trigger snippet and opening the dialog that way

Reply:
Yes. I'll change it.